packages/mui-lab/src/LoadingButton/LoadingButton.js

```diff
@@ -132,6 +132,14 @@ const LoadingButtonLoadingIndicator = styled('div', {
       position: 'relative',
       right: -10,
     }),
+  ...(ownerState.loadingPosition === 'start' &&
+      ownerState.size === 'small' && {
+      left: 10,
```

Comment:
The issue was about misalignment with respect to the vertical center.

Reply:
You can reproduce the issue with small sized button, for normal and large buttons it is working fine

Reply:
Yes, when size is `small`, the text is slightly misaligned due to `line-height` property. Adding left or right padding is irrelevant to this issue.

Reply:
line-height issue is there for text inside button, but the issue here is the space between icon and text

![image](https://user-images.githubusercontent.com/11643426/154266350-d6815ba1-f58e-448d-a61c-18b189fc3b49.png)

Reply:
for normal and large buttons, alignment is perfect

![image](https://user-images.githubusercontent.com/11643426/154266847-abfb1697-197e-4a72-a258-44aecf772940.png)

Reply:
I see. Thanks, I misunderstood the issue.

Would you add unit tests for the change you made?